evaluate/report/csv.go

```diff
@@ -121,3 +181,164 @@ func WriteEvaluationRecord(resultPath string, model model.Model, language langua
+	}
+	defer evaluationFile.Close()
+
+	reader := csv.NewReader(evaluationFile)
+
+	// Check if the evaluation CSV header is correct.
+	if header, err := reader.Read(); err != nil {
+		return nil, pkgerrors.Wrap(err, "file is empty")
+	} else if !reflect.DeepEqual(header, EvaluationHeader()) {
+		return nil, pkgerrors.WithStack(pkgerrors.Errorf("expected header %+v\nfound header %+v", EvaluationHeader(), header))
+	}
+
+	// Read the raw records from the evaluation CSV file.
+	records, err := reader.ReadAll()
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+
+	// Convert the raw records into assessments that can be easily manipulated.
+	evaluationRecords = EvaluationRecords{}
+	for _, record := range records {
+		evaluationRecord, err := convertRawRecordToEvaluationRecord(record)
+		if err != nil {
+			return nil, err
+		}
+		evaluationRecords = append(evaluationRecords, evaluationRecord)
+	}
+
+	return evaluationRecords, nil
+}
+
+// convertRawRecordToEvaluationRecord converts a raw CSV record into an evaluation record.
+func convertRawRecordToEvaluationRecord(raw []string) (record *EvaluationRecord, err error) {
+	assessments := metrics.NewAssessments()
+
+	modelID := raw[0]
+	modelName := raw[1]
+	modelCost, err := strconv.ParseFloat(raw[2], 64)
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+
+	languageID := raw[3]
+
+	coverage, err := strconv.ParseUint(raw[7], 10, 64)
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+	assessments[metrics.AssessmentKeyCoverage] = coverage
+
+	filesExecuted, err := strconv.ParseUint(raw[8], 10, 64)
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+	assessments[metrics.AssessmentKeyFilesExecuted] = filesExecuted
+
+	generateTestsForFileCharacterCount, err := strconv.ParseUint(raw[9], 10, 64)
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+	assessments[metrics.AssessmentKeyGenerateTestsForFileCharacterCount] = generateTestsForFileCharacterCount
+
+	processingTime, err := strconv.ParseUint(raw[10], 10, 64)
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+	assessments[metrics.AssessmentKeyProcessingTime] = processingTime
+
+	responseCharacterCount, err := strconv.ParseUint(raw[11], 10, 64)
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+	assessments[metrics.AssessmentKeyResponseCharacterCount] = responseCharacterCount
+
+	responseNoError, err := strconv.ParseUint(raw[12], 10, 64)
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+	assessments[metrics.AssessmentKeyResponseNoError] = responseNoError
+
+	responseNoExcess, err := strconv.ParseUint(raw[13], 10, 64)
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+	assessments[metrics.AssessmentKeyResponseNoExcess] = responseNoExcess
+
+	responseWithCode, err := strconv.ParseUint(raw[14], 10, 64)
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+	assessments[metrics.AssessmentKeyResponseWithCode] = responseWithCode
```

Comment:
Please just do a `for` loop since all the values are `uint`. The order of the keys can be obtained from the assessment package. It's the same order as it was written. That way it is also futureproof in case we change the order of keys or add new ones.